fcrepo-exts/fcrepo-camel-toolbox
# Fcrepo 3919 - Make solr endpoint credentials configurable

**Make solr endpoint credentials configurable**
* * *

**JIRA Ticket**: https://fedora-repository.atlassian.net/jira/software/c/projects/FCREPO/issues/FCREPO-3919?filter=myopenissues

# What does this Pull Request do?
Adds configurable username and password properties for HTTP Basic Auth when indexing Solr documents into an auth protected Solr instance

# How should this be tested?

* Configure a Solr instance with HTTP Basic Auth enabled https://solr.apache.org/guide/8_1/basic-authentication-plugin.html
* Set properties solr.username and solr.password according to the documentation 
* Test solr indexing endpoint to confirm that authentication works for a set of valid credentials and fails for invalid credentials

# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo/committers
